### PR TITLE
feat(cli): use cjs dist stub in package

### PR DESCRIPTION
**Pull request by pimlie**
## Types of changes

- [ ] Bug fix (a non-breaking change which fixes an issue)
- [x] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Description

As discussed on discord, the cli package could use a cjs stub so we dont have to use `-r esm` all the time. The other packages dont need this because they dont provide a point of entry
and are thus not directly invoked during development. They are either called through the cli or through jest when testing (and jest didnt seem to like all the esm stubs).

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly. (PR: #)
- [ ] I have added tests to cover my changes (if not applicable, please state why)
- [x] All new and existing tests are passing.

### CODE REVIEW

File: scripts/dev [hunk 1]
```diff
@@ -5,6 +5,26 @@ import consola from 'consola'
 
 import Package from './package.js'
 
+const useCjs = new Set([
+  '@nuxt/cli'
+])
+
+const stub = {
+  es: `export * from '../src/index'`,
```
Comment (pi0) [inline scripts/dev]: Why don't just add a shebang to the first line all files:
```
#! /usr/bin/env node -r esm
```

File: test/unit/cli.test.js [hunk 2]
```diff
@@ -9,7 +9,7 @@ const rootDir = resolve(__dirname, '..', 'fixtures/cli')
 const url = route => 'http://localhost:' + port + route
 
 const nuxtBin = resolve(__dirname, '../../packages/cli/bin/nuxt.js')
-const spawnNuxt = (command, opts) => spawn('node', ['-r', 'esm', nuxtBin, command, rootDir], opts)
+const spawnNuxt = (command, opts) => spawn('node', [nuxtBin, command, rootDir], opts)
```
Comment (pi0) [inline test/unit/cli.test.js]: We can directly spawn `nuxtBin` when having shebang :)

Comment (pi0) [reply test/unit/cli.test.js]: PS: I think there was another related unit test too

File: scripts/dev (hunk 1 above)
Comment (pimlie) [reply scripts/dev]: Does that work for you on a mac? When I do that on linux it says:
```
/usr/bin/env: ‘node -r esm’: No such file or directory
```

File: test/unit/cli.test.js (hunk 2 above)
Comment (pimlie) [reply test/unit/cli.test.js]: Thanks, will change that. Better indeed to keep testing as close to real world usage as much as possible :)

File: scripts/dev (hunk 1 above)
Comment (pi0) [reply scripts/dev]: Don't want to add more complexity

**Review by pi0 — APPROVED** [review]

**Comment by pimlie** [comment]
@pi0 The failing test on azure is again a timing issue in the component tests I added for `nuxt-loading.vue`. Maybe I should redo those tests to not rely on waitFor, but would appreciate some suggestions how to do that differently ;)

Maybe we should skip those component tests for now on Azure?

**Pull request merged**